core/utils.js

```diff
@@ -719,3 +739,43 @@ Blockly.utils.wrapToText_ = function(words, wordBreaks) {
   }
   return text.join('');
 };
+
+/**
+ * Check if 3D transforms are supported by adding an element
+ * and attempting to set the property.
+ * @return {boolean} true if 3D transforms are supported
+ */
+Blockly.is3dSupported = function() {
+  if (Blockly.cache3dSupported_ !== null) {
+    return Blockly.cache3dSupported_;
+  }
+  // CC-BY-SA Lorenzo Polidori
+  // https://stackoverflow.com/questions/5661671/detecting-transform-translate3d-support
+  if (!window.getComputedStyle) {
+    return false;
+  }
+
+  var el = document.createElement('p'),
+    has3d,
+    transforms = {
+      'webkitTransform': '-webkit-transform',
+      'OTransform': '-o-transform',
+      'msTransform': '-ms-transform',
+      'MozTransform': '-moz-transform',
+      'transform': 'transform'
+    };
+
+  // Add it to the body to get the computed style.
+  document.body.insertBefore(el, null);
```

Comment:
Oh neat. Thanks. Not sure it is applicable here since this isn't really replacing anything.  It is inserting the note into the dom to check whether it can perform certain 3d stuff, then removing it.